### PR TITLE
only set stream header on streamed output

### LINKED ISSUE #1528: webui broken by get fix
the get fix broke the webui

---

https://botbot.me/freenode/ipfs/2015-07-28/?msg=45757898&page=3

```
<slothbag> I just rolled back to the commit before http get fix and it works
```

### PR DESCRIPTION
**Pull request by whyrusleeping**
fixes #1528 

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: commands/http/handler.go [hunk 1]
```diff
@@ -182,6 +174,11 @@ func sendResponse(w http.ResponseWriter, req cmds.Request, res cmds.Response) {
 		h.Set(contentLengthHeader, strconv.FormatUint(res.Length(), 10))
 	}
 
+	if _, ok := res.Output().(io.Reader); ok {
+		mime = ""
```
Comment (jbenet) [inline commands/http/handler.go]: preserve the comment, else we won't know why the hell we're setting `mime = ""` in a few weeks.

Comment (whyrusleeping) [inline commands/http/handler.go]: but i didnt like that comment... :(

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 98559d0e0bddd62d4c478df0e67e34e5dbef9c4e
  * Invalid signoff. Commit message must end with **License: MIT
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23)

---
This message was auto-generated by https://gitcop.com

**Comment by whyrusleeping** [comment]
omg gitcop, shut up. If i want to use githubs online editor to write code, I can do so. I am an adult.

**Comment by jbenet** [comment]
@whyrusleeping then put the trailer in manually. don't blame gitcop, gitcop's right.

File: commands/http/handler.go [hunk 2]
```diff
@@ -182,6 +174,14 @@ func sendResponse(w http.ResponseWriter, req cmds.Request, res cmds.Response) {
 		h.Set(contentLengthHeader, strconv.FormatUint(res.Length(), 10))
 	}
 
+	if _, ok := res.Output().(io.Reader); ok {
+		// we don't set the Content-Type for streams, so that browsers can MIME-sniff the type themselves
+		// we set this header so clients have a way to know this is an output stream
+		// (not marshalled command output)
+		mime = ""
+		h.Set(streamHeader, "1")
```
Comment (jbenet) [inline commands/http/handler.go]: so _only_ set the header when it is an io.Reader and mime is off?

is this right?. why doesnt a `chan interface{}` have `"StreamHeader: 1"` too? (it is written out like a stream). what does streamHeader mean exactly?

Comment (whyrusleeping) [inline commands/http/handler.go]: streamheader means that youre getting an arbitrary stream of data. `chan interface{}` means youre getting a series of json objects.

**Pull request merged**